src/test/groovy/pl/allegro/tech/discovery/consul/recipes/watch/health/HealthServiceInstancesWatcherIntTest.groovy

```diff
@@ -50,19 +57,46 @@ class HealthServiceInstancesWatcherIntTest extends Specification {
         }
 
         when: "unhealthy service instance is registered"
-        consulCluster.registerUnhealthyServiceInstance("my-service", "dc1", "node1-dc1")
+        consulCluster.registerUnhealthyServiceInstance("my-service", GIVEN_DC, GIVEN_NODE)
 
         and: "healthy service instance is registered"
-        consulCluster.registerHealthyServiceInstance("my-service", "dc1", "node1-dc1", ["tag1", "tag2"])
+        consulCluster.registerHealthyServiceInstance("my-service", GIVEN_DC, GIVEN_NODE, ["tag1", "tag2"])
 
         then: "watcher caught new state with only healthy service instance"
         new PollingConditions(timeout: 10).eventually {
             latestState.head().instances.size() == 1
             def instance = latestState.head().instances.first()
             instance.serviceId != null
-            instance.serviceAddress == "localhost"
-            instance.servicePort == 1234
+            instance.serviceAddress.get() == "localhost"
+            instance.servicePort.get() == 1234
```

Comment:
I don't like using `Optional#get` without checking if value is presnt. This way when something work wrong in tests we will get NPE. I suggest to compare it to `Optional` or checking presents before getting value.
Similar with other assertion with `Optional`

Reply:
changed check now